src/geo/geohash.clj

```diff
@@ -33,21 +36,11 @@
                               (.getMinLat box)
                               (.getMaxLat box)))))
 
-(defn northern-neighbor
```

Comment:
Type hinting the above GeoHash to shape might be good. I imagine this maybe used rather often.

Actually generally speaking when dealing with a lot of Java like this, it's good to turn on reflection warnings and see where you're taking that hit.